cpp/oneapi/dal/backend/memory.hpp

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
```

Comment:
In the backend we can have more aggressive inlining (but no in the public API, because of BC problems), that's why those functions are defined in headers.

Moreover on the kernel side, we typically work with the queue, but not with `data_parallel_policy`.

Reply:
I'd suggest making `backend` version for host allocation routines too - does it make sense?

Reply:
I think, yes. The reason they are not there is that ndaray does not support host allocations yet. I suggest to implement this later once we need them, having DPC++ is more critical.

I expect in all cases when we need even a host memory in GPU algorithms, we'll allocate shared.